test/unit/org/apache/cassandra/index/sai/cql/LuceneAnalyzerTest.java

```diff
@@ -575,6 +575,19 @@ public void testInvalidQueryOnNumericColumn() throws Throwable
         .isInstanceOf(InvalidRequestException.class);
     }
 
+    @Test
+    public void testLegacyEqQueryOnNormalizedTextColumn()
+    {
+        createTable("CREATE TABLE %s (id int PRIMARY KEY, val text)");
+        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex' WITH OPTIONS = {'ascii': 'true', 'case_sensitive': 'false', 'normalize': 'true'}");
+        waitForIndexQueryable();
+
+        execute("INSERT INTO %s (id, val) VALUES (1, 'Aaą')");
+        flush();
+
+        assertEquals(1, execute("SELECT * FROM %s WHERE val = 'aaa'").size());
```

Comment:
Can we do this before and after flushing? Maybe inside `beforeAndAfterFlush(CheckedFunction runnable)`?

Reply:
done